tools/infrasys/src/s3.rs

```diff
@@ -176,8 +179,11 @@ pub async fn upload_file(
     prefix: &str,
     file_path: &Path,
 ) -> Result<()> {
-    let s3_client =
-        S3Client::new(Region::from_str(region).context(error::ParseRegionSnafu { what: region })?);
+    let config = aws_config::from_env()
```

Comment:
Not sure if this throws everything for a loop, but is there a way to generate the `aws_config` in a single place and pass a reference around rather than continuously getting it?

Reply:
That makes sense to me, but I'll have to look at it a bit more closely since this is @rpkelly's commit.